pinecone/client.go

```diff
@@ -273,6 +273,104 @@ func (c *Client) DeleteIndex(ctx context.Context, idxName string) error {
 	return nil
 }
 
+// ConfigureIndex is used to [scale a pods-based index] up or down by changing the size of the pods or the number of
+//replicas.
+//
+// Parameters:
+//  - name: The name of the index to configure.
+//  - pods: (Optional) The pod size to scale the index to (e.g. for a "p1" pod type,
+//  you could pass "p1.x2" to scale your index to the "x2" size,
+//  or you could pass "p1.x4" to scale your index to the "x4" size, and
+//  so forth.
+//  - replicas: (Optional) The number of replicas to scale the index to.
+//   This is capped by the maximum number of replicas allowed in your Pinecone project. To configure this number,
+//   go to [app.pinecone.io], select your project, and configure the maximum number of pods.
+//
+// Note: You can only scale an index up, not down. If you want to scale an index down,
+// you must create a new index with the desired configuration.
+//
+// Returns a ConfigureIndexResponse object, which contains the new configuration of the index, or an error.
+//
+// Example for a pods-based index originally configured with 1 "p1" pod of size "x2" and 1 replica:
+//  // To scale the size of your pods from "x2" to "x4":
+//   _, err := pc.ConfigureIndex(ctx, "my-index", "p1.x4", nil)
+//   if err != nil {
+//       fmt.Printf("Failed to configure index: %v\n", err)
+//   }
+//
+//  // To scale the number of replicas:
+//   _, err := pc.ConfigureIndex(ctx, "my-index", nil, 4)
+//   if err != nil {
+//       fmt.Printf("Failed to configure index: %v\n", err)
+//   }
+//
+//  // To scale both the size of your pods and the number of replicas:
+//   _, err := pc.ConfigureIndex(ctx, "my-index", "p1.x4", 4)
+//   if err != nil {
+//       fmt.Printf("Failed to configure index: %v\n", err)
+//   }
+//
+// [scale a pods-based index]: https://docs.pinecone.io/guides/indexes/configure-pod-based-indexes
+// [app.pinecone.io]: https://app.pinecone.io
+func (c *Client) ConfigureIndex(ctx context.Context, name string, pods *string,
```

Comment:
`pods` may be a little confusing given that fields means something else when creating an index. Although I'm realizing now we don't provide a `pods` value in `CreatePodIndexRequest`.

Still, we use `PodType` there and I think we should do the same here. I know it's a positional arg, but just for clarity in the code here. I know we then assign things to `podType` further down, but initially when I was reading the function I was a bit confused.

Reply:
Okay, np, I'll change to `podType`! (I hate how we call it `type` b/c technically it's `size`, but yeah best not to confuse ppl who're used to our other SDKs)